.github/workflows/test.yml

```diff
@@ -0,0 +1,21 @@
+name: Run tests
+
+on:
+  pull_request:
+    branches: [master]
+
+jobs:
+  tests:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Python 3.8
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.8
+      - name: Upgrade pip
+        run: python -m pip install --upgrade pip setuptools
+      - name: Install dependencies
+        run: pip install black isort mypy pylint pytest tutor types-docutils types-PyYAML types-setuptools
```

Comment:
This command will install the latest version of each of these dependencies. If any of these dependencies releases breaking changes (such as a new default check) then it could cause the PR checks to start failing.

For this reason, Tutor (and most Open edX repos) use a system where they list their versionless dependencies in a ".in" file, and then generate a list of specifically-versioned dependency pins into a ".txt" file. Requirements are then installed from the .txt file (e.g., `pip install -r requirements/development.txt`). In order to keep the versions fresh, the pins in the .txt files are periodically regenerated from the versionless requirements in the .in files using the `make upgrade-requirements` command. [OEP-18](https://open-edx-proposals.readthedocs.io/en/latest/best-practices/oep-0018-bp-python-dependencies.html) dives deep into the details if you're interested.

Can you update this PR to use such a system? You can use Tutor itself as an example. You'll need to:

* Put these testing requirements into [requirements/dev.in](https://github.com/overhangio/tutor/blob/master/requirements/dev.in)
* Add an [`upgrade-requirements` target to the Makefile](https://github.com/overhangio/tutor/blob/f4dc508c1913b818f69259125493109cea6144f8/Makefile#L18) which runs pip-compile on dev.in
* Run `make upgrade-requirements` in order to generate [requirements/dev.txt](https://github.com/overhangio/tutor/blob/master/requirements/dev.txt)
* Install requirements from requirements/dev.txt in this test workflow.

Most repositories have "base.in" and "docs.in" as well, but I think you can ignore those for this PR.

Let me know if you have any questions. Happy to help clarify.

Reply:
> I think this is a great improvement!

Glad to hear that!

I agree with you and I mentioned having such a file [here](https://github.com/overhangio/tutor-mfe/issues/151#issuecomment-1742621681) also, but @regisb suggested keeping all plugins and cookiecutter itself as lean as possible. If everyone agrees, I'm fine with making updates to the PR. I just want to ensure that everyone is on board with the changes.

Reply:
Haha, I should have figured you were already thinking about this.

Curious to hear what you think @regisb .

Reply:
For what it's worth, I think pinning the requirements is as important with plugins as with any other package.

Reply:
@regisb I assume you missed this thread.

Reply:
Sorry about that!

I suggest we pin the requirements in the CI file. Then, once we are happy with how CI works, we will move it out of the plugin repo to a dedicated repository, such that it can be reused across plugins. That way, we will have both the benefit of a reproducible environment, and a lean plugin repository.

Reply:
OK, so I used the pinned versions of `dev.txt` in tutor (there was no pinned version for `pytest`, so I used the latest).